README.md

```diff
@@ -377,20 +405,46 @@ val consumer = connection.newPullConsumer[Bytes](???, ???)
 
 
 // receive "up to" 100 deliveries
-val deliveries: Future[Seq[Option[DeliveryWithHandle[Future, Bytes]]]] = Future.sequence { (1 to 100).map(_ => consumer.pull()) }
+val deliveries: Future[Seq[PullResult[Future, Bytes]]] = Future.sequence { (1 to 100).map(_ => consumer.pull()) }
 
 // do your stuff!
 
 ???
 
-// "handle" all deliveries
-val handleResult: Future[Unit] = deliveries.flatMap(s => Future.sequence(s.flatten.map(_.handle(DeliveryResult.Ack))).map(_ => Unit))
+// "handle" all deliveries, ignore failures and "empty queue" results
+val handleResult: Future[Unit] = deliveries.flatMap(s => Future.sequence(s.flatMap(_.toOption).map(_.handle(DeliveryResult.Ack))).map(_ => Unit))
 
 consumer.close()
 connection.close()
 
 ```
 
+#### Pull-Consumer parsing failure
+
+It may happen the delivery is not parsable as your `MyClass`. `ParsingFailureAction` will be used for handling such message and you will get
```

Comment:
I would prefer to remove this feature - it's not IMHO required because you can achieve the same behavior just by processing the `PullResult.MalformedContent` when calling `pull`. You have to handle this case anyway...